src/server/services/toeristische-verhuur.ts

```diff
@@ -23,10 +39,86 @@ export interface ToeristischeVerhuurRegistratie {
   street: string | null;
 }
 
+export interface TransformedVakantieverhuur extends Vakantieverhuur {
+  isPast: boolean;
+  duration: number | null;
+}
+
+export interface TransformedVakantieverhuurAfmelding
+  extends VakantieverhuurAfmelding {
+  isPast: boolean;
+  duration: number | null;
+}
+
+export interface TransformedVakantieverhuurVergunningaanvraag
+  extends VakantieverhuurVergunningaanvraag {
+  isPast: boolean;
+  duration: number | null;
+}
+
+export interface TransformedBBVergunning extends BBVergunning {
+  isPast: boolean;
+  duration: number | null;
+}
+
+export type VakantieverhuurVergunningen =
+  | Vakantieverhuur
+  | VakantieverhuurAfmelding
+  | BBVergunning
+  | VakantieverhuurVergunningaanvraag;
+
+export type TransformedVakantieverhuurVergunningen =
+  | TransformedVakantieverhuur
+  | TransformedVakantieverhuurAfmelding
+  | TransformedBBVergunning
+  | TransformedVakantieverhuurVergunningaanvraag;
+
 export interface ToeristischeVerhuurRegistratiesSourceData {
   content: ToeristischeVerhuurRegistratie[];
 }
 
+export const daysLeftInCalendarYear = (items: Vakantieverhuur[]): number => {
+  const itemsThisYear = items?.filter((item) => {
+    return item.dateStart ? isCurrentYear(item.dateStart) : undefined;
+  });
+  return (
+    itemsThisYear?.reduce((a, b) => {
+      if (b.dateEnd ? isCurrentYear(b.dateEnd) : undefined) {
+        return (
+          a -
+          (b.dateEnd && b.dateStart
+            ? formatDurationBetweenDates(b.dateEnd, b.dateStart)
+            : 0)
+        );
+      } else {
+        const daysTillEndOfYear =
+          b.dateEnd && b.dateStart
+            ? formatDurationBetweenDates(b.dateEnd, b.dateStart)
+            : 0;
+        return a - daysTillEndOfYear;
+      }
+    }, DAYS_LEFT_TO_RENT) ?? DAYS_LEFT_TO_RENT
+  );
+};
+
+export function transformVergunningenToVerhuur(
+  vergunningen: VakantieverhuurVergunningen[] | null
+): TransformedVakantieverhuurVergunningen[] | null {
```

Comment:
Waarom return je hier `null`? Is een lege array niet beter?

Reply:
Kun je niet beter iets doen als: `if (!Array.isArray(vergunningen)) { return []; }`